dask/array/gufunc.py

```diff
@@ -212,6 +212,13 @@ def apply_gufunc(func, signature, *args, **kwargs):
     vectorize: bool, keyword only
         If set to ``True``, ``np.vectorize`` is applied to ``func`` for
         convenience. Defaults to ``False``.
+    reassign_vectorize_dtypes: bool, keyword only
+        If set to ``True``, after ``np.vectorize`` is applied to ``func``,
+        then the output dtypes are assigned to func.otypes.  This overcomes
+        a bug in numpy.vectorize that mangles dtypes.  This only matters if
+        ``vectorize`` is ``True``.  This should most likely be set to ``True``
+        but is defaults to ``False`` for backward compatibility.  Defaults
+        to ``False``.
```

Comment:
There is a ux cost to adding new options to methods. I suspect that no users are relying on the current broken behavior, and we could apply this fix universally with no option (my preference). Thoughts?